### PR TITLE
Finish Search Screen

### PR DESCRIPTION
**Pull request by AarushiSingh214**
## Overview

Implemented the search screen so the user can search for a product and the results appear. The results will filter based off what the user enters into the search bar. The user can now also see the history of their searches

## Changes Made

Created a component called SearchBar that implements the search bar and then created another screen called SearchScreen that integrates the search bar so that the accurate results appear when the user searches for a product. 

## Screenshots (delete if not applicable)

<details>

  <summary>Screen Shot Name</summary>

https://user-images.githubusercontent.com/70385317/163426074-1a378ae9-b75e-4b50-b731-f7a4ab943d43.MP4

</details>

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -24,16 +24,17 @@ export default function App() {
...
-        console.log("Google SignIn", "FAILURE", result);
-      }
-    });
+    //   if (type == "success") {
+    //     console.log("Google SignIn", "SUCCESS", result);
+    //     setSignIn(true);
+    //   } else {
+    //     console.log("Google SignIn", "FAILURE", result);
+    //   }
+    // });
```
Comment (alanna-zhou) [inline App.tsx:36]: is there a reason why commented code is being committed haha

File: components/ProductList.tsx [hunk 2]
```diff
@@ -38,6 +45,13 @@ export function ProductList({ count, filter, data, navigation }) {
       />
     ) : null;
   };
+
+  function compareItem(item) {
```
Comment (alanna-zhou) [inline components/ProductList.tsx:49]: wouldn't `itemContainsSearchKeyWord` be a better func name

File: screens/HomeScreen.tsx [hunk 3]
```diff
@@ -13,38 +13,91 @@ import { DATA } from "../data/product";
...
-        color={"#808080"}
-        theme={{ colors: { accent: "white" } }}
-      />
+      {!isSearch && (
+        <ButtonBanner
+          count={count}
+          setCount={setCount}
+          data={FILTER}
+        //   modalVisible={undefined}
+        //   setModalVisible={undefined}
```
Comment (alanna-zhou) [inline screens/HomeScreen.tsx:78]: oopsies

**Review by alanna-zhou — COMMENTED** [review]
i dont know anything about react but im just snooping around hahahaha

File: components/HistoryList.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+} from "react-native";
+import React from "react";
+import { TouchableWithoutFeedback } from "react-native-gesture-handler";
+
+export default function HistoryList({ searchHistory, searchSubmit }) {
+  /**
+   *
+   * @param {state} count - state passed in representing which button is getting clicked, should be less than equal to len(data) -1
+   * @param {setState} setCount - setState method passed in to change the current count when new button get clicked
+   * @param {list} data - a list of button names
```
Comment (Haichen-Wang) [inline components/HistoryList.tsx:16]: I don't see these as params

File: navigation/LinkingConfiguration.ts [hunk 5]
```diff
@@ -41,7 +40,9 @@ const linking: LinkingOptions<RootStackParamList> = {
 
       Modal: 'modal',
       NotFound: '*',
-      NewPost:'NewPost'
+      NewPost:'NewPost',
+                    ProductHome: 'product/:id', // allow product details to be opened with a link specifying the product id
```
Comment (Haichen-Wang) [inline navigation/LinkingConfiguration.ts:44]: nit: indentation

**Review by Haichen-Wang — COMMENTED** [review]
Nicely done! I left a few comments. I'm not approving this PR rn but will review once my comments are answered!

**Pull request merged**